include/zephyr/shell/shell.h

```diff
@@ -234,6 +234,82 @@ struct shell_static_entry {
 	uint8_t padding[Z_SHELL_STATIC_ENTRY_PADDING];
 };
 
+/**
+ * @brief Shell structured help descriptor.
+ *
+ * @details This structure provides an organized way to specify command help
+ * as opposed to a free-form string. This helps make help messages more
+ * consistent and easier to read.
+ */
+struct shell_cmd_help {
+	/* @cond INTERNAL_HIDDEN */
+	uint32_t magic;
+	/* @endcond */
+	const char *description; /*!< Command description */
+	const char *usage;       /*!< Command usage string */
+};
+
+/**
+ * @cond INTERNAL_HIDDEN
+ */
+
+/**
+ * @brief Magic number used to identify the beginning of a structured help
+ * message when cast to a char pointer.
+ */
+#define SHELL_STRUCTURED_HELP_MAGIC 0x86D20BC4
+
+/**
+ * @endcond
+ */
+
+/**
+ * @brief Check if help string is structured help.
+ *
+ * @param help Pointer to help string or structured help.
+ * @return true if help is structured, false otherwise.
+ */
+static inline bool shell_help_is_structured(const char *help)
+{
+	const struct shell_cmd_help *structured = (const struct shell_cmd_help *)help;
+
+	return structured != NULL && structured->magic == SHELL_STRUCTURED_HELP_MAGIC;
+}
+
+#if CONFIG_SHELL_HELP
```

Comment:
```suggestion
#if defined(CONFIG_SHELL_HELP) || defined(_DOXYGEN_)
```

Reply:
Dang I had fixed it but then messed up in a rebase.
 Thanks

Reply:
done